protoc_plugin/lib/src/enum_generator.dart

```diff
@@ -164,27 +162,29 @@ class EnumGenerator extends ProtobufContainer {
               ]);
         }
       }
-      out.println();
 
-      out.println('static const $coreImportPrefix.List<$classname> values ='
-          ' <$classname> [');
-      for (final val in _canonicalValues) {
-        final name = dartNames[val.name];
-        out.println('  $name,');
-      }
-      out.println('];');
       out.println();
-
       out.println(
           'static final $coreImportPrefix.Map<$coreImportPrefix.int, $classname> _byValue ='
           ' $protobufImportPrefix.ProtobufEnum.initByValue(values);');
       out.println('static $classname? valueOf($coreImportPrefix.int value) =>'
           ' _byValue[value];');
-      out.println();
 
+      out.println();
+      out.println('@$coreImportPrefix.override');
+      out.println('final $coreImportPrefix.int value;');
+      out.println();
+      out.println('@$coreImportPrefix.override');
+      out.println('final $coreImportPrefix.String name;');
+      out.println();
+      out.println('const $classname(this.value, this.name);');
+      out.println();
       out.println(
-          'const $classname._($coreImportPrefix.int v, $coreImportPrefix.String n) '
-          ': super(v, n);');
+          "/// Returns this enum's [name] or the [value] if names are not"
+          ' represented.');
+      out.println('@$coreImportPrefix.override');
+      out.println("$coreImportPrefix.String toString() => name == '' ? "
+          'value.toString() : name;');
```

Comment:
I had to generate a "custom" `toString` method for each enum, since the default enum `toString` takes priority over the one from `ProtobufEnum`.

This method is designed to be 100% compatible, but it might be better to return `Enum.name` (`super.toString()`) instead of `name`.